File: cpp_solutions/945.cpp
class Solution {
public:
    int minIncrementForUnique(vector<int>& nums) {
        int res = 0;
        sort(nums.begin(), nums.end());
        int n = nums.size();
        int criteria = -1;
        for (int i = 0; i < n; ++i) {
            if (nums[i] > criteria) {
                criteria = nums[i];
            }
            else {
                criteria++;
                res += criteria - nums[i];
            }
        }
        return res;
    }
};

File: rust_solutions/945.rs
impl Solution {
    pub fn min_increment_for_unique(mut nums: Vec<i32>) -> i32 {
        nums.sort_unstable();
        let mut res = 0;
        let mut current = -1;
        let n = nums.len();
        for i in 0..n {
            if nums[i] > current {
                current = nums[i];
            }
            else {
                current += 1;
                res += (current - nums[i]);
                nums[i] = current;
            }
        }
        res
    }
}

